### PR TITLE
Allow stopping api server when watch command is stopped

### LINKED ISSUE #923: API server should restart if `watch` command fails
Hi, with the current `--watch` implementation on the API server, if error occurs in the watch goroutine, process just stops leaving API server running without performing any work. You need to manually restart the server or start watch command by the API call in order to have it running again.
Also `watch` states are not clearly visible, for example if you run backup server in kubernetes, pod will be treated as healthy even if it does not perform expected functionality.

I think it would make sense to be able to configure API server that it is considered healthy only when watch goroutine is running and stop the API server when watch command fails. 
I made a pull request to address this [issue](https://github.com/Altinity/clickhouse-backup/pull/922)

### PR DESCRIPTION
**Pull request by tadus21**
Hi, with the current `--watch` implementation on the API server, if error occurs in the `watch` goroutine, process just stops leaving API server running without performing any work and you need to manually restart the server or `start` watch command by the API call in order to have it running again. 

I think it would make sense to stop API server if `watch` command failed so I am proposing a change that allows running `watch` command as a main API process and once its stopped, whole server is stopped as well. 

Added this feature under a new `watch_is_main_process` config argument so it is backwards compatible and will not introduce new behaviour in the tool without explicitly enabling it.

fix https://github.com/Altinity/clickhouse-backup/issues/923

Does this change make sense and I could expect it to be introduced in the system?

### CODE REVIEW

**Comment by Slach** [comment]
build failed
could check `make clean build/linux/amd64/clickhouse-backup` locally?

File: pkg/config/config.go
Comment (Slach) [inline pkg/config/config.go]: why you remove this line?

**Review by Slach — CHANGES_REQUESTED** [review]

File: pkg/config/config.go
Comment (tadus21) [reply pkg/config/config.go]: oh sorry, my ide for some reason decided that this import is not used, restored it back.

**Comment by Slach** [comment]
for tesflows pass
```
rm -rf ./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot
cp ./test/testflows/.env.example ./test/testflows/.env
```
edit `./test/testflows/.env`

run tests
```
bash -xe ./test/testflows/run.sh
```

commit changed cli.py.cli.snapshot

**Review by Slach — APPROVED** [review]

**Comment by Slach** [comment]
All looks good now, thanks for contribution

small step before merge
could you run pull latest master? i'm upgraded github action build.yaml to allow Cleanup step works properly

**Comment by tadus21** [comment]
> All looks good now, thanks for contribution
> 
> small step before merge could you run pull latest master? i'm upgraded github action build.yaml to allow Cleanup step works properly

Thanks for helping with this. I think because I rebased master I need to get a new workflow approval in order to merge this

**Pull request merged**
